### DISCUSSION TITLE
Bemf 0.000 v

### DISCUSSION THREAD

**Discussion by WMP** [nliaudat/esp32_8ch_motor_shield#41] [General]
I have all bemf all time 0.000 v, maybe is problem with my motors? 
![1000002032](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/95bba9c1-cf68-46ce-aeef-cf473f468650)
I use V57 rev 1.48 . 

And I see that my board keep restarting, I haven't more that 15 minutes uptime

**Comment by nliaudat** [comment]
Hi, 
Hard to say what's wrong cause you must provide more infos. 
Is your connection with hassio good ? If hassio is enabled and no communication, the board reboots after 15 min. 

board.yml
```
# Enable Home Assistant API
api:
  reboot_timeout: 15min
```

What voltage do you have between GND and CH1-2 and other at running and at endstop ?
![bemf_custom_resistor](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/6d01f556-4c99-408d-b04e-0e8d4da41505)


If you want help, you must have search yourself before a bit ...

**Reply by WMP** [reply]
Can I check this voltage on motor not mounted on valve? I haven't added this board to HA yet...

**Comment by nliaudat** [comment]
If you haven't mout the motor, there will be no resistance nor endstop. The BEMF value of 0.0 is right

**Reply by nstadigs** [reply]
I guess @WMP has connected the motors to the board, but he hasn't mounted the motors to the valves.

**Reply by WMP** [reply]
![PXL_20230903_183152980 MP](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/129e0717-f1a6-4e0c-8bcf-88fbe1909821)

**Comment by WMP** [comment]
I asked in the context of testing - my floor heating is behind the cabinet and I don't really want to pull the cabinet back to check the voltage. In addition, I use actuators in it after the return, which do work though.  I know I have the same situation in the other floor heating, and it is easier to access and there is a new actuator.

In order not to have to test the board in the floor heating, I will buy another new actuator and some cheap m30x1.5 valve and check how the situation is on the 3rd board. I am using a GPIO22 together with a 4.7 kOhm resistor to which I have a DS18B20 connected - maybe this is the problem?

**Reply by nliaudat** [reply]
You can disconnect 1 actuator and set it up to your board. 
You should have a reading of "moving" values and 1 endstop when the screw that acts as the plunger is at the endstop on the motor side. (you have Homematic actuators).

GPIO22 is good as it's in free pins : https://github.com/nliaudat/esp32_8ch_motor_shield/blob/main/extension.md

**Comment by nliaudat** [comment]
If you have one, check with another esp32-devkit. Some very cheap board have no buildin ADC calibration and may report incorrect values

**Comment by WMP** [comment]
Hmmm, my voltage metter show 0.000v...

1. Check if my voltage metter works:
![PXL_20230906_205532234](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/65f21bdf-ba87-4fcc-bab6-87aa60551ea2)
So, works. I have good connected GND and 5v.

2. Then i press Open or Close, my motor works, my valve moving but i see all time only 0.000v on ESP webpage or in my voltage metter:
![PXL_20230906_205435403](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/c722d6bc-9484-4416-bc1a-4d6b87cfbc99)
![PXL_20230906_205326262 MP](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/cab64418-64eb-457a-863b-cbec941250c9)
I see 0.000v on all BEMF channels.

**Reply by WMP** [reply]
Here i have max 0.105 before my motor made a clicking noise:
![CM230906-231327006](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/d7bf0885-e470-4c53-9371-eb7872122a94)

**Comment by nliaudat** [comment]
I suspect a bad LM324 module .... Your value of  0.105V in CH1-BEMF is good for endstop. 

Can you check that the CH1-BEMF value is the same at "12In" ? (or direct in the pin if the connector is too small)
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/14aea8cd-1b51-4bcb-88b7-f7ebb58f669f)

And also check "12Out" which must be 6x "12In". 

Regards

**Reply by WMP** [reply]
I have https://jlcpcb.com/partdetail/Hgsemi-LM324AMTR/C2688471 or https://jlcpcb.com/partdetail/Hgsemi-LM324MTR/C194286

I think i cannot test anymore because... :
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/9fb7aaf9-f471-4c84-854c-2dfebbb27945)
Is possible to bought this item or do you have 3d printer project for this? 

So my test results can be wrong:
"12In" - from 0.012 to 0.056 before clicking noise
(this foot in LM324) - from 0.012 to 
"12Out" -  flapping from 0.001 to 0.003

**Reply by nliaudat** [reply]
![2023-09-08 17 48 32](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/96aaf578-fdf7-4e85-933e-b3043bba3ab1)

**Reply by WMP** [reply]
This is printed on 3d?

**Reply by WMP** [reply]
Ahhh, I see... metal screw or something

**Reply by WMP** [reply]
I have this: 

LM324MTR    LM324    SOIC-14_L8.7-W3.9-P1.27-LS6.0-BL    C194286
1.27-1*8P母    OPAMP-H1    HDR-TH_8P-P1.27-V-F    C691120

**Comment by nliaudat** [comment]
If your opamp module if bad, a quick fix can be to link the BEMF inputs to the board like this : 

![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/b328996b-1e97-4f90-b03c-895382f6074a)

The previous version with no opamp were like that and were working, but not with all kind of motors. You have homematic motors and I run my board in that configuration for years.

**Reply by WMP** [reply]
After i connect only 1channel bemf to CH1|2 i have on both pins 0.000:
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/1b62c5db-40f1-43ab-a4df-9b4e9904123e)

**Reply by nliaudat** [reply]
And you can read BEMF values in the ESP32 board ....

**Reply by WMP** [reply]
Yes, I have 0.000, but maybe this is my broken motor...

**Comment by nliaudat** [comment]
Drill 1mm diam, insert a nail and cut it the right lenght.

**Comment by WMP** [comment]
I fix my motor, and i insert it onto my valve. I see how vale moving up and down. But i stil have all time 0.000v.
![PXL_20230912_212548268 MP~2](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/78a5e6af-36d0-4ab0-a0bb-ce9675d8a024)


When i try to change firmware code to: https://github.com/nliaudat/floor-heating-controller/blob/main/sensor_adc.yaml#L61 i see all time 0.008v in ESPhome, but in voltage metter i have only 0.000v.

**Reply by WMP** [reply]
In addition, during testing the plastic to which I attached the nail ans bent....

![PXL_20230912_212903211 MP~2](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/45381/03bc2a43-b7a0-4506-8f3e-ef9261015431)

**Reply by WMP** [reply]
With yellow cabel what i use to  connect BEMF,  in 12in and 12out i have 0.000v. 
Without yellow cabel, i have 0.013v in CH1-BEMF and "12In" and 0.002v in 12Out.

**Reply by WMP** [reply]
If you want, i can send you one of my board to debug this.

**Reply by nliaudat** [reply]
Please first isolate 230V parts top & bottom with tape like in https://github.com/nliaudat/esp32_8ch_motor_shield/blob/main/imgs/v58_rev_1.49.jpg

![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/b5fe6cb3-ad4a-4825-98a5-c2d2b7986a9d)

**Reply by WMP** [reply]
Ok, but why?

**Reply by nliaudat** [reply]
Cause it's a bit dangerous isn't it ?

**Reply by WMP** [reply]
It is, but power is low.

**Reply by nliaudat** [reply]
It's not ! It's AC 230V

**Reply by WMP** [reply]
Yes, this is voltage. But power on this voltage is low, I dont measure how low, but I think this is under few Wats.

**Reply by WMP** [reply]
But of course more that 48v is dangerous.

**Reply by nliaudat** [reply]
`  >100 mA and it's a cardiac arrest`

https://www.firstaidpro.com.au/electric-shock-first-aid-survive-240-volts

**Reply by WMP** [reply]
Assuming that the current draw on the 5v side is maximum - 5w, or 1A, and the efficiency of the power supply is 70%, it comes out that at 230v there is 0.031 A of current.

**Reply by nstadigs** [reply]
This has nothing to do with what's on the rest of the board. L and N on the 230v side is directly connected to the outlet, so it's the same as putting your fingers in the wall outlet.

**Reply by nliaudat** [reply]
And if you touch 230V pin with your finger, your feets are the ground, what is the A drawn by your heart ?

**Reply by nliaudat** [reply]
The discussion is closed. If you don't isolate the 230V part, I won't answer any more queries. Send a photo as proof

**Comment by nliaudat** [comment]
If you have a second board, please try to cut off the resistor as told in https://github.com/nliaudat/esp32_8ch_motor_shield/blob/main/hack.md . You must cut CH1 and CH2

Have you tried another ESP32 devkit ? some are broken.

**Reply by WMP** [reply]
I have 5 free boards, I can check with other esp first...  I have few without external antena, i can check this.

**Reply by WMP** [reply]
I haven't insulated the AC yet, and for now I'm not touching the board :). However, I think I have found the problem. In the documentation https://docs.espressif.com/projects/esp-idf/en/v4.4.2/esp32/api-reference/peripherals/adc.html#_CPPv425adc1_config_channel_atten14adc1_channel_t11adc_atten_t it says that usually esp32 measures voltage from 100mV, and according to my meter I have a maximum of 105mV. This is close enough to the minimum limit that I think it is not measuring this voltage for some reason. Another esp32-s2 already measures the voltage from 0mV. Do you think that using esp32-s2 can solve the problem?
